scico/denoiser.py

```diff
@@ -38,19 +43,21 @@ def bm3d(x: JaxArray, sigma: float, is_rgb: bool = False):
     Returns:
         Denoised output.
     """
+    if not have_bm3d:
+        raise RuntimeError("Package bm3d is required for use of this function." "")
 
     if is_rgb is True:
         bm3d_eval = tunibm3d.bm3d_rgb
     else:
         bm3d_eval = tunibm3d.bm3d
 
     if np.iscomplexobj(x):
-        raise TypeError(f"BM3D requries real-valued inputs, got {x.dtype}")
+        raise TypeError(f"BM3D requires real-valued inputs, got {x.dtype}")
 
     # Support arrays with more than three axes when the additional axes are singletons.
     x_in_shape = x.shape
 
-    if x.ndim < 2:
+    if isinstance(x.ndim, tuple) or x.ndim < 2:
```

Comment:
What is the idea behind checking x.ndim is tuple?

Reply:
Unless the type annotations are incorrect. Do we allow BlockArrays here?

Reply:
We don't allow BlockArrays, but in the absence of the additional test for a tuple, this test fails to raise the appropriate exception if one is passed as the argument.